GoogleForCreators/web-stories-wp
# Keyboard Shortcuts: Accessibility updates

## Context

While working on the help center and checklist popups, I noticed that the keyboard shortcut menu - which sits right next to these popups - varies a bit and can't be navigated with a keyboard. 

## Summary

Uses aXe dev tools in tandem with aXe `toHaveNoViolations` karma test on keyboard shortcut pop up to make sure it meets a11y standards. While I was at it, I synced the styles with the other two popups and cleaned up some unnecessary code now that we've got updated styles available in the design system.

Updates include: 
- Keyboard shortcuts now use `secondary` color instead of `tertiary` to pass AA color contrast requirements
- Scrollable content is now also scrollable with keyboard arrows, if you use a keyboard on the menu that section is given a focus border - just like our other focusable spots.
- Toggle button is same style as other two toggle buttons in secondary menu 

## Relevant Technical Choices

- There were a lot of roles getting applied where they're superfluous - such as list and listitem and group. We can rely on the description lists themselves to give the component context and meaning. I cleaned this up and made sure the `dl`s are organized right (had to flip the layout with css so that the html hierarchy remains in tact). 
- Cleaned up the focus style so that it is achieved in the same manner as help center is.
- Reuse the shared popup, removed the isolated one used just for this. 

| Main | This Branch | 
| --- | --- |
| ![main](https://user-images.githubusercontent.com/10720454/128779822-be677fd3-502d-476e-9d53-aeba2a467f61.png) | ![updated](https://user-images.githubusercontent.com/10720454/128779831-6a94e943-d88a-4ba9-af2e-ddac76a13918.png) |

(this is kind of deceiving, the updated version in this PR is complaining about color contrast - but everything passes. So i think for some reason it's picking up border color.) 
| | |
| --- | --- | 
| <img width="684" alt="Screen Shot 2021-08-09 at 2 33 01 PM" src="https://user-images.githubusercontent.com/10720454/128779911-d938b96c-b183-49ed-b360-a1fb19813a81.png"> | <img width="654" alt="Screen Shot 2021-08-09 at 2 32 13 PM" src="https://user-images.githubusercontent.com/10720454/128779912-421282f4-e146-4f0c-a388-72e07f68ea0e.png"> |

## To-do

- I am pretty sure that aria labels on `kbd`s are unnecessary, if we can confirm that we can remove those - they're causing a warning on aXe dev tools. Worst case, they are just redundant, not causing any harm. 

## User-facing changes

- Cursor uses should notice no changes
- Keyboard users should now be able to scroll through the keyboard shortcut popup instead of being trapped on the close button.
- Keyboard toggle button looks like the one used by the help center and checklist

## Testing Instructions

Use both cursor and keyboard to navigate the keyboard shortcut menu. Make sure there are no regressions and that the content is scrollable using both a keyboard and the mouse.
If you have access, run some aXe dev tools against the popup, notice there's no high priority issues with the popup. 

### QA

- Verify new toggle for keyboard shortcuts menu works the same as before (clicking once opens, clicking again closes. Same goes for 'enter')




This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #7912
